plutus-playground-server/test/Playground/InterpreterSpec.hs

```diff
@@ -0,0 +1,59 @@
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Playground.InterpreterSpec
+    ( tests
+    ) where
+
+import           Control.Monad.Except         (runExceptT)
+import           Data.Aeson                   (toJSON)
+import qualified Data.Text                    as Text
+import           Language.Haskell.Interpreter (SourceCode (SourceCode))
+import qualified Ledger.Ada                   as Ada
+import           Playground.Interpreter       (mkExpr, mkRunScript)
+import           Playground.Types             (Evaluation (Evaluation, program, sourceCode, wallets),
+                                               Expression (AddBlocks), PlaygroundError,
+                                               SimulatorWallet (SimulatorWallet))
+import           Test.Tasty                   (TestTree, testGroup)
+import           Test.Tasty.HUnit             (assertEqual, testCase)
+import           Wallet.Emulator.Types        (Wallet (Wallet))
+
+tests :: TestTree
+tests = testGroup "Playground.Interpreter" [mkRunScriptTest]
+
+mkRunScriptTest :: TestTree
+mkRunScriptTest =
+    testGroup
+        "mkRunScript"
+        [ testCase "Should match a simple template." $ do
+              expr :: Either PlaygroundError String <-
+                  runExceptT $
+                  mkExpr
+                      (Evaluation
+                           { sourceCode = sourceCode1
+                           , wallets =
+                                 [ SimulatorWallet (Wallet 1) (Ada.toValue 5)
+                                 , SimulatorWallet (Wallet 2) (Ada.toValue 10)
+                                 ]
+                           , program = toJSON [AddBlocks 2, AddBlocks 4]
+                           })
+              assertEqual
+                  ""
+                  (Right $
+                   Text.stripEnd $
+                   Text.unlines
+                       [ "module Main where"
+                       , ""
+                       , "foo :: Int"
+                       , "foo = 5"
+                       , ""
+                       , ""
+                       , "main :: IO ()"
+                       , "main = printJson $ stage endpoints \"[{\\\"blocks\\\":2,\\\"tag\\\":\\\"AddBlocks\\\"},{\\\"blocks\\\":4,\\\"tag\\\":\\\"AddBlocks\\\"}]\" \"[{\\\"simulatorWalletBalance\\\":{\\\"getValue\\\":[[{\\\"unCurrencySymbol\\\":\\\"\\\"},[[{\\\"unTokenName\\\":\\\"\\\"},5]]]]},\\\"simulatorWalletWallet\\\":{\\\"getWallet\\\":1}},{\\\"simulatorWalletBalance\\\":{\\\"getValue\\\":[[{\\\"unCurrencySymbol\\\":\\\"\\\"},[[{\\\"unTokenName\\\":\\\"\\\"},10]]]]},\\\"simulatorWalletWallet\\\":{\\\"getWallet\\\":2}}]\""
```

Comment:
I'd just encode the value with Aeson right here instead of having the literal JSON object in the source code

Reply:
Good idea!